lib/moped/connection.rb

```diff
@@ -34,9 +34,10 @@ def alive?
     # @since 1.0.0
     def connect
       @sock = if !!options[:ssl]
-        Sockets::SSL.connect(host, port, timeout)
+        ssl_options = options.reject { |k, v| !k.to_s.start_with?('ssl') }
+        Sockets::SSL.connect(host, port, timeout, ssl_options)
       else
-        Sockets::TCP.connect(host, port, timeout)
+        Sockets::TCP.connect(host, port, timeout, {})
```

Comment:
If the other initializers / methods are made optional, then I think you could skip having to pass the options explicitly here